client/request_handler.go

```diff
@@ -22,7 +30,16 @@ func NewRequestHandler(client BaseClient) *RequestHandler {
 	}
 }
 
-func (c *RequestHandler) sendRequest(method string, rawURL string, data url.Values,
+func NewRequestHandlerWithContext(client BaseClient) *RequestHandlerWithContext {
```

Comment:
Not sure you need a separate struct for this honestly, just add a new method/methods on the RequestHandler and call it from everything.